hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/WorldLedgers.java

```diff
@@ -133,6 +155,140 @@ public boolean defaultKycStatus(final TokenID tokenId) {
+                feeScheduleCandidate.ifPresent(k -> info.setFeeScheduleKey(asKeyUnchecked(k)));
+
+                final var pauseCandidate = token.pauseKey();
+                pauseCandidate.ifPresentOrElse(
+                        k -> {
+                            info.setPauseKey(asKeyUnchecked(k));
+                            info.setPauseStatus(tokenPauseStatusOf(token.isPaused()));
+                        },
+                        () -> info.setPauseStatus(TokenPauseStatus.PauseNotApplicable));
+
+                if (token.hasAutoRenewAccount()) {
+                    info.setAutoRenewAccount(token.autoRenewAccount().toGrpcAccountId());
+                    info.setAutoRenewPeriod(
+                            Duration.newBuilder().setSeconds(token.autoRenewPeriod()));
+                }
+
+                info.addAllCustomFees(token.grpcFeeSchedule());
+
+                return Optional.of(info.build());
+            } catch (Exception unexpected) {
+                log.warn(
+                        "Unexpected failure getting info for token {}!",
+                        readableId(tokenId),
+                        unexpected);
+                return Optional.empty();
+            }
+        }
+    }
+
+    public Optional<TokenNftInfo> infoForNft(final NftID target, final ByteString ledgerId) {
+        if (staticEntityAccess != null) {
+            return staticEntityAccess.infoForNft(target);
+        } else {
+            final var tokenId = EntityNum.fromTokenId(target.getTokenID());
+            final var targetKey =
+                    NftId.withDefaultShardRealm(tokenId.longValue(), target.getSerialNumber());
+            if (!nftsLedger.contains(targetKey)) {
+                return Optional.empty();
+            }
+            final var targetNft = nftsLedger.getImmutableRef(targetKey);
+            var accountId = targetNft.getOwner().toGrpcAccountId();
+
+            if (WILDCARD_OWNER.equals(accountId)) {
+                var merkleToken = tokensLedger.getImmutableRef(target.getTokenID());
+                if (merkleToken == null) {
+                    return Optional.empty();
+                }
+                accountId = merkleToken.treasury().toGrpcAccountId();
+            }
+
+            final var spenderId = targetNft.getSpender().toGrpcAccountId();
+
+            final var info =
+                    TokenNftInfo.newBuilder()
+                            .setLedgerId(ledgerId)
+                            .setNftID(target)
+                            .setAccountID(accountId)
+                            .setCreationTime(targetNft.getCreationTime().toGrpc())
+                            .setMetadata(ByteString.copyFrom(targetNft.getMetadata()))
+                            .setSpenderId(spenderId)
+                            .build();
+            return Optional.of(info);
+        }
+    }
+
+    public List<CustomFee> tokenCustomFees(final TokenID tokenId) {
```

Comment:
We can rename the method to `infoForTokenCustomFees` and make it return `Optional`, so that we are compliant with the other info methods.